packages/core/utils/src/traverse/query-populate.ts

```diff
@@ -202,15 +199,15 @@ const populate = traverseFactory()
 
     if (isObject(value)) {
       const { components } = attribute;
-
       const newValue = {};
 
       // Handle legacy DZ params
       let newProperties: unknown = omit('on', value);
 
       for (const componentUID of components) {
         const componentSchema = strapi.getModel(componentUID);
-        newProperties = await recurse(visitor, { schema: componentSchema, path }, newProperties);
+        const properties = await recurse(visitor, { schema: componentSchema, path }, value);
+        newProperties = merge(newProperties, properties);
```

Comment:
I guess this is more of a manual test for now, but what happens if you make a query where one relation should be stripped out (don't have the permissions to read the data), and one can be populated?

e.g.
- `A` has a dynamic zone with two components (C1 has a relation to `R1`, and `C2` has a relation to `R2`)
- You have the permission to read `A` & `R2`, but not `R1` (from the rest API, using U&P)
- You make a query trying to populate everything in the DZ.
- Is `R1` still removed successfully? Or is it kept? (because we're not sending the same object anymore, could have an impact)

I guess that it should still work but somehow I'm a little confused.

Reply:
From my testing it seems to be working, it's not impacting sanitization, I don't see how it would either.
you only get the stuff you have permission to see

Reply:
Yes, it makes sense, thanks for confirming. Might just be my brain triggering false alarms.

I'll do some manual QA before merging anyway.